tsdb/mapper.go

```diff
@@ -216,7 +216,7 @@ func (lm *LocalMapper) Open() error {
 					// No data exists for this key.
 					continue
 				}
-				seriesTags := lm.shard.index.series[key].Tags
```

Comment:
Sure, then I presume the nil check in the new function is because the
function might be called from other places in the future. That is my
question.

On Thursday, August 20, 2015, Paul Dix <notifications@github.com> wrote:

> In tsdb/mapper.go
> <https://github.com/influxdb/influxdb/pull/3749#discussion_r37537279>:
>
> > @@ -216,7 +216,7 @@ func (lm *LocalMapper) Open() error {
> >  					// No data exists for this key.
> >  					continue
> >  				}
> > -				seriesTags := lm.shard.index.series[key].Tags
>
> no, it was a data race because it was accessing a shared map without a
> lock. The series is guaranteed to be in the map when you get to this point.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/influxdb/influxdb/pull/3749/files#r37537279>.
>